frontend/public/components/row-filter.jsx

```diff
@@ -71,11 +68,34 @@ class CheckBoxes_ extends React.Component {
       } catch (ignored) {
         // ignore
       }
-
       this.setState({selected}, () => this.applyFilter());
     }
   }
 
+  toggle(event, itemId) {
+    event.preventDefault();
+    const selected = _.xor(this.state.selected, [itemId]);
+    this.setQueryParameters(selected);
+  }
+
+  selectAll() {
+    const selected = _.map(this.props.items, 'id');
+    this.setQueryParameters(selected);
+  }
+
+  countNumItems() {
+    const selectedCheckboxes = _.map(this.state.selected, (id) => {
+      return this.props.numbers[id] || 0;
+    });
+    const numSelectedItems = _.sum(selectedCheckboxes);
+    const numTotalItems = _.sum(_.values(this.props.numbers));
```

Comment:
I just realized this isn't correct... It's possible for some items to not match any filters. I think we need to pass the total count into the component.